server/src/domain/library/library.service.ts

```diff
@@ -572,6 +571,36 @@ export class LibraryService extends EventEmitter {
     return JobStatus.SUCCESS;
   }
 
+  // Check if an asset is has no file or is outside of import paths, marking it as offline
+  async handleOfflineCheck(job: ILibraryOfflineJob): Promise<JobStatus> {
+    const asset = await this.assetRepository.getById(job.id);
+
+    if (!asset || asset.isOffline) {
+      // We only care about online assets, we exit here if offline
+      return JobStatus.SKIPPED;
+    }
+
+    const exists = await this.storageRepository.checkFileExists(asset.originalPath, R_OK);
+
+    let existsInImportPath = false;
+
+    for (const importPath of job.importPaths) {
+      if (asset.originalPath.startsWith(importPath)) {
+        existsInImportPath = true;
+        break;
+      }
+    }
```

Comment:
```suggestion
    const path = job.importPaths.find((path) => asset.originalPath.startsWith(path));
```
And then down below check if `path && exists`

Reply:
Done, thanks